code/datums/components/item_two_handed.dm

```diff
@@ -0,0 +1,339 @@
+	if(!wielded || !user)
+		return
+
+	// wield update status
+	wielded = FALSE
+	UnregisterSignal(user, COMSIG_MOB_SWAP_HANDS)
+	SEND_SIGNAL(parent, COMSIG_TWOHANDED_UNWIELD, user)
+	if(on_unwield_callback)
+		on_unwield_callback.Invoke(user)
+
+	// update item stats
+	var/obj/item/parent_item = parent
+	if(force_multiplier)
+		parent_item.force /= force_multiplier
+	else if(force_unwielded)
+		parent_item.force = force_unwielded
+
+	// update the items name to remove the wielded status
+	var/sf = findtext(parent_item.name, " (Wielded)", -10) // 10 == length(" (Wielded)")
+	if(sf)
+		parent_item.name = copytext(parent_item.name, 1, sf)
+	else
+		parent_item.name = "[initial(parent_item.name)]"
+
+	// Update icons
+	parent_item.update_icon()
+	if(user.get_item_by_slot(ITEM_SLOT_BACK) == parent)
+		user.update_inv_back()
+	else
+		user.update_inv_hands()
+
+	// Show message if requested
+	if(show_message)
+		if(iscyborg(user))
+			to_chat(user, "<span class='notice'>You free up your module.</span>")
+		else if(require_twohands)
+			to_chat(user, "<span class='notice'>You drop [parent].</span>")
+		else
+			to_chat(user, "<span class='notice'>You are now carrying [parent] with one hand.</span>")
+
+	// Play sound if set
+	if(unwieldsound)
+		playsound(parent_item.loc, unwieldsound, 50, TRUE)
+
+	// Remove the object in the offhand
+	var/obj/item/offhand/offhand_item = user.get_inactive_held_item()
+	if(offhand_item && istype(offhand_item))
+		offhand_item.unwield()
+
+/**
+ * on_attack triggers on attack with the parent item
+ */
+/datum/component/two_handed/proc/on_attack(obj/item/source, mob/living/target, mob/living/user)
+	if(wielded && attacksound)
+		var/obj/item/parent_item = parent
+		playsound(parent_item.loc, attacksound, 50, TRUE)
+
+/**
+ * on_update_icon triggers on call to update parent items icon
+ *
+ * Checked in the order listed, first found is used
+ * Invokes the Icon Update Callback with wield status
+ * Updates the icon using icon_prefix if set
+ */
+/datum/component/two_handed/proc/on_update_icon(datum/source)
+	if(icon_update_callback)
+		icon_update_callback.Invoke(wielded)
+	else if(icon_prefix)
+		var/obj/item/parent_item = parent
+		if(parent_item)
+			parent_item.icon_state = "[icon_prefix][wielded]"
+
+/**
+ * try_wield tries to wield the item
+ */
+/datum/component/two_handed/proc/try_wield(datum/source, mob/user)
+	wield(user)
+
+/**
+ * try_unwield attempts to unwield the item
+ */
+/datum/component/two_handed/proc/try_unwield(datum/source, mob/user, show_message=TRUE)
+	unwield(user, show_message)
+
+/**
+ * on_swap_hands Triggers on swapping hands, blocks swap if the other hand is busy
+ */
+/datum/component/two_handed/proc/on_swap_hands(mob/user, obj/item/held_item)
+	if(!held_item)
+		return
+	var/datum/component/two_handed/comp_twohand = held_item.GetComponent(/datum/component/two_handed)
+	if(comp_twohand && comp_twohand.wielded)
```

Comment:
You don't need to GetComponent, just block the swap `if(held_item == parent)`

Reply:
Will be fixed in next update